cns/requestcontroller/kubecontroller/crdtranslator.go

```diff
@@ -1,19 +1,80 @@
 package kubecontroller
 
 import (
+	"net"
+	"strconv"
+	"strings"
+
 	"github.com/Azure/azure-container-networking/cns"
 	nnc "github.com/Azure/azure-container-networking/nodenetworkconfig/api/v1alpha"
 )
 
-// CRDStatusToNCRequest translates a crd status to network container request
+// CRDStatusToNCRequest translates a crd status to createnetworkcontainer request
 func CRDStatusToNCRequest(crdStatus *nnc.NodeNetworkConfigStatus) (*cns.CreateNetworkContainerRequest, error) {
-	//TODO: Translate CRD status into network container request
-	//Mat will pick up from here
-	return nil, nil
+	var (
+		createNCRequest   cns.CreateNetworkContainerRequest
+		nc                nnc.NetworkContainer
+		secondaryIPConfig cns.SecondaryIPConfig
+		ipSubnet          cns.IPSubnet
+		ipAssignment      nnc.IPAssignment
+		prefixLength      int
+	)
+
+	createNCRequest.SecondaryIPConfigs = make(map[string]cns.SecondaryIPConfig)
+
+	for _, nc = range crdStatus.NetworkContainers {
+		createNCRequest.NetworkContainerid = nc.ID
+		createNCRequest.PrimaryInterfaceIdentifier = nc.PrimaryIP
```

Comment:
This is not correct. NC Primary ip will be saved in createNCRequest.IPConfiguration